### PR TITLE
LMDB storage: clarify disk space usage limits, minor wording changes

### LINKED ISSUE #2034: Improve communication about LMDB disk space usage
Hello!

Users have expressed concerns about LMBD disk space usage. There is an understanding this is for optimization, as mentioned in the docs. However, it is not stated when LMDB will start reusing this reserved disk space. Kero mentioned in this [discussion](https://github.com/meilisearch/meilisearch/discussions/3156#discussioncomment-4262368) that LMDB would reuse the space through compaction execution, which occurs each time a snapshot is created. 

It'd be fantastic to mention this in the FAQ or the Snapshot page.

Related issue:

https://github.com/meilisearch/meilisearch/discussions/3156
https://github.com/meilisearch/meilisearch/discussions/3138

Thanks

### PR DESCRIPTION
**Pull request by guimachiavelli**
Closes #2034

### CODE REVIEW

File: learn/advanced/storage.md [hunk 1]
```diff
@@ -10,11 +10,9 @@ The database location can change depending on a number of factors, such as wheth
...
 
-The storage engine Meilisearch uses under the hood is a [Lightning Memory-Mapped Database](http://www.lmdb.tech/doc/) (LMDB for short). LMDB is a transactional key-value store written in C that was developed for OpenLDAP and has ACID properties.
-
-We chose LMDB after extensive testing with LMDB itself, [Sled](https://github.com/spacejam/sled) and [RocksDB](https://rocksdb.org/). We chose LMDB because it provided us with the best combination between performance and stability.
+Our storage engine Meilisearch is called [Lightning Memory-Mapped Database](http://www.lmdb.tech/doc/) (LMDB for short). LMDB is a transactional key-value store written in C that was developed for OpenLDAP and has ACID properties. Though we considered other options, such as [Sled](https://github.com/spacejam/sled) and [RocksDB](https://rocksdb.org/), we chose LMDB because it provided us with the best combination between performance and stability.
```
Comment (dureuill) [inline learn/advanced/storage.md:15]: performance, stability, and features.

**Review by dureuill — APPROVED** [review]
Thanks, the changes summarize things well! Looks good to me :heart:

**Review by maryamsulemani97 — APPROVED** [review]
🏃‍♂️

**Review by oluademola — APPROVED** [review]
LGTM! Thank you very much @guimachiavelli 🙌

**Review by maryamsulemani97 — APPROVED** [review]
🤹

**Comment by guimachiavelli** [comment]
bors merge

**Pull request merged**